### ISSUE TITLE
disk.IOCounters() fails in virtual windows

### ISSUE THREAD

**Issue by sskserk** [shirou/gopsutil#1094]
**Describe the bug**
Executing of disk.IOCounters() in Virtual Windows (WEMU or VMWare) player results in error:
"The system cannot find the path specified" 

**To Reproduce**
Compile the following code with the CLI: **GOOS=windows go build -tags windows -o file.exe file.go**

```go
	_, err := disk.IOCounters()
	if err != nil {
		panic(err)
	}
```

**Expected behavior**
Returned disk io counters

**Additional context**

- The code is compiled on Ubuntu 20.04 and executed on virtual Windows Server 2019 running in VMWare Player. Host is Ubuntu 20.04 
- Debugging of the source revealed that the path "\\.\c:" cannot be found. The failure is observed in this piece of code https://github.com/shirou/gopsutil/blob/ec072d99db0fecd5663e7127a3bd39c958522a23/disk/disk_windows.go#L167

**Comment by Lomanic** [comment]
What surprises me here is that if the path is really missing [the loop's step should be skipped above](https://github.com/shirou/gopsutil/blob/ec072d99db0fecd5663e7127a3bd39c958522a23/disk/disk_windows.go#L157-L159), is `C:` mounted and accessible? What happens [if you replace these lines](https://github.com/shirou/gopsutil/blob/ec072d99db0fecd5663e7127a3bd39c958522a23/disk/disk_windows.go#L168-L170) by [these that check for this specific error](https://github.com/shirou/gopsutil/blob/ec072d99db0fecd5663e7127a3bd39c958522a23/disk/disk_windows.go#L158-L162), do you get some results back or all disks are skipped?

I won't be able to test in VMWare Player unfortunately, but I implemented this part on Virtualbox though, there shouldn't be much difference in my opinion.

**Comment by mixmind** [comment]
Hi, it dont collect information after adding check that you proposed but fixed error. Yes, C: is mounted and accesible, but discovered path is incorrect...
Had you received data in Virtualbox?

**Comment by xgfone** [comment]
I met with the trouble.

1. Build the app with `github.com/shirou/gopsutil/v3/disk` on `Windows 10 64bit`.
2. Run it in the Windows 10, and it's OK.
3. Run it in the Virtual Windows 10 (QEMU 2.8), and it's OK.
4. Run it in the Virtual Windows 2008~2019 Standard/DataCenter (QEMU 2.8). It does not work and reports the error `The system cannot find the file specified.`.

**Comment by xgfone** [comment]
@Lomanic 
- For Windows 7 & 10, it is OK. 
- For all Windows Server 2008~2019, it does not work and reports the error.

**Comment by xgfone** [comment]
refer to #883, run `diskperf -y`, then it's OK.

**Comment by Lomanic** [comment]
Thanks for linking the two issues together and finding a workaround @xgfone!

That probably explains why I couldn't reproduce #883 on Win7 x32 as it seems to only affect Server editions.

As I asked in https://github.com/shirou/gopsutil/issues/883#issuecomment-635504791, can you tell us if you can reproduce again after you run `diskperf.exe -N`? Can you check if `diskperf.exe -Y` creates the registry entry mentioned in https://github.com/shirou/gopsutil/issues/883#issuecomment-635504791 so we could automate this in gopsutil?

**Comment by xgfone** [comment]
@Lomanic I have done the tests as follow.
- For Windows 10 64bit, 
  1. `diskperf.exe -N` has no effect, and `disk.IOCounters()` returns always success.
  2. Execute  `diskperf.exe -Y`, and check `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\PartMgr`, no `EnableCounterForIoctl`.
- For Windows Server 2008~2019, 
    1. Call `disk.IOCounters()`, and return the error `The system cannot find the file specified.`.
    2. Execute the command `diskperf.exe -Y`, then call `disk.IOCounters()` again, and it is OK.
    3. Check `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\PartMgr`, `EnableCounterForIoctl` is set to `1`.
    4. Execute the command `diskperf.exe -N`, then call `disk.IOCounters()` again, and it returns the error above.
    5. Again check `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\PartMgr`, `EnableCounterForIoctl` has been deleted.
    6. Execute the command `diskperf.exe -Y`, and retstart the system.
    7. After restarting, login and call `disk.IOCounters()`, it is still OK.

In conclusion, Windows 7&10 does not use `diskperf.exe` and `EnableCounterForIoctl` to manage the disk performance counter, and enable it by default. However, Windows Servers use `diskperf.exe` and `EnableCounterForIoctl` to control DPC, and disable it by default.

For Windows Servers, we maybe enable it automatically when importing the sub-package `disk` in the initialization function `init`.

**Comment by CHN-STUDENT** [comment]
@xgfone hi, so if we need to monitor windows disk io on windows server. 
We need run `diskperf.exe -Y` on cmd command?
